io/genbank/genbank.go

```diff
@@ -146,222 +144,314 @@ func getFeatureSequence(feature Feature, location Location) (string, error) {
-			meta.Keywords = joinSubLines(splitLine, subLines)
-		case "SOURCE":
-			meta.Source, meta.Organism = getSourceOrganism(splitLine, subLines)
-		case "REFERENCE":
-			meta.References = append(meta.References, getReference(splitLine, subLines))
-			continue
-		case "FEATURES":
-			features = getFeatures(subLines)
-		case "ORIGIN":
-			sequence.Sequence = getSequence(subLines)
-			sequenceBreakFlag = true
-		default:
-			if quickMetaCheck(line) {
-				key := strings.TrimSpace(splitLine[0])
-				meta.Other[key] = joinSubLines(splitLine, subLines)
-			}
-		}
+	genbankSlice, err := ParseMultiNth(r, -1)
 
+	if err != nil {
+		return []Genbank{}, err
 	}
 
-	// add meta to genbank struct
-	sequence.Meta = meta
+	return genbankSlice, err
+}
 
-	// add features to annotated sequence with pointer to annotated sequence in each feature
-	for _, feature := range features {
-		_ = sequence.AddFeature(&feature)
+// ParseMultiNth takes in a reader representing a multi gbk/gb/genbank file and parses the first n records into a slice of Genbank structs.
+func ParseMultiNth(r io.Reader, count int) ([]Genbank, error) {
+	scanner := bufio.NewScanner(r)
+	var sequence = Genbank{}
 
-	}
+	// Create meta struct
+	var meta = Meta{}
 
-	return sequence, nil
-}
+	meta.Other = make(map[string]string)
 
-// Build builds a GBK string to be written out to db or file.
-func Build(sequence Genbank) ([]byte, error) {
-	var gbkString bytes.Buffer
-	locus := sequence.Meta.Locus
-	var shape string
+	// Create features struct
+	var features = []Feature{}
+	var feature = Feature{}
+	feature.Attributes = make(map[string]string)
 
-	if locus.Circular {
-		shape = "circular"
-	} else if locus.Linear {
-		shape = "linear"
-	}
+	// Metadata setup
+	var metadataTag string
+	var metadataData []string //this stutters but will remain to make it easier to batch rename variables when compared to metadataTag.
 
-	fivespace := generateWhiteSpace(subMetaIndex)
+	// Feature setup
+	var newLocation bool
+	var quoteActive bool
 
-	// building locus
-	locusData := locus.Name + fivespace + locus.SequenceLength + " bp" + fivespace + locus.MoleculeType + fivespace + shape + fivespace + locus.GenbankDivision + fivespace + locus.ModificationDate
-	locusString := "LOCUS       " + locusData + "\n"
-	gbkString.WriteString(locusString)
+	var attribute string
+	var attributeValue string
 
-	// building other standard meta features
-	definitionString := buildMetaString("DEFINITION", sequence.Meta.Definition)
-	gbkString.WriteString(definitionString)
+	// Sequence setup
+	reg, _ := regexp.Compile("[^a-zA-Z]+")
+	var sequenceBuilder strings.Builder
 
-	accessionString := buildMetaString("ACCESSION", sequence.Meta.Accession)
-	gbkString.WriteString(accessionString)
+	// Sequence breaking
+	parseStep := "metadata" // we can be parsing metadata, features, or sequence
 
-	versionString := buildMetaString("VERSION", sequence.Meta.Version)
-	gbkString.WriteString(versionString)
+	genbankStarted := false
```

Comment:
@Koeng101 was this supposed to be used somewhere? The flag is set to true but it doesn't seem to ever be checked.